### PR TITLE
renamed NSAlertStyle constants

### PR DESCRIPTION
**Pull request by SamSchott**
The old constants `NSWarningAlertStyle` etc have been deprecated in macOS 10.13 in favour of `NSAlertStyleWarning` etc (see [AppKit docs](https://developer.apple.com/documentation/appkit/nsalertstyle?language=objc)). The values of the constants have not changed so we maintain backward compatibility.

### CODE REVIEW

File: src/cocoa/toga_cocoa/dialogs.py [hunk 1]
```diff
@@ -66,7 +64,7 @@ def error(window, title, message):
 def stack_trace(window, title, message, content, retry=False):
     alert = NSAlert.alloc().init()
     alert.icon = window.app.icon._impl.native
-    alert.setAlertStyle_(NSCriticalAlertStyle)
+    alert.setAlertStyle_(NSAlertStyle.Critical)
```
Comment (goanpeca) [inline src/cocoa/toga_cocoa/dialogs.py]: Is there a missing `.value` here?

Comment (SamSchott) [reply src/cocoa/toga_cocoa/dialogs.py]: There is. Just corrected.

Comment (freakboy3742) [reply src/cocoa/toga_cocoa/dialogs.py]: Are you sure the `.value` is required? It doesn't *hurt* to have it - but Rubicon should convert Python Enums into their underlying value types... if it doesn't, then that's a bug in Rubicon.

Comment (SamSchott) [reply src/cocoa/toga_cocoa/dialogs.py]: I didn't know that Rubicon did that :)

Comment (goanpeca) [reply src/cocoa/toga_cocoa/dialogs.py]: Me neither :-p

**Comment by freakboy3742** [comment]
There's clearly something going on with Windows CI... it looks like it might be related to a release of pip 20.

**Comment by freakboy3742** [comment]
The CI failure appears to be due to the pip 20.0 release; I've just merged a fix that seems to address the problem.

**Comment by SamSchott** [comment]
Thanks!

**Pull request merged**